packages/engine/lib/error/src/report.rs

```diff
@@ -367,6 +367,28 @@ impl<T> Report<T> {
         self.frames().find_map(Frame::downcast_ref::<A>)
     }
 
+    /// Returns the current context of the `Report`.
+    ///
+    /// If the user want to get the latest context, `current_context` can be called. If the user
+    /// wants to handle the error, the context can then be used to directly access the context's
+    /// type. This is only possible for the latest context as the Report does not have multiple
+    /// generics as this would either require variadic generics or a workaround like tuple-list.
+    ///
+    /// This is one disadvantage of the library in comparison to plain Errors, as in these cases,
+    /// all context types are known.
+    #[must_use]
+    #[allow(clippy::missing_panics_doc)]
+    pub fn current_context(&self) -> &T
+    where
+        T: Any,
+    {
+        // Panics if no context is attached which should only happen when using the hooks.
```

Comment:
What does this mean about the hooks?

Reply:
When using hooks, we use `Report<()>` to abstract over the context. When calling `current_context()` this would try to downcast to `()`.

Reply:
I think we should reword the comment because it could come across as saying it panics when using the hooks.

> Panics is there isn't an attached context. It should be only be possible to create a Report without a context when using the hooks, and this method shouldn't be able to be called then.

Reply:
I improved the documentation and also make this function only callable, if `T: Context`.